core/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java

```diff
@@ -82,11 +83,17 @@
 
     private final BigArrays bigArrays;
     private final ScriptService scriptService;
+    private final Client client;
 
-    public SearchPhaseController(Settings settings, BigArrays bigArrays, ScriptService scriptService) {
+    public SearchPhaseController(Settings settings, BigArrays bigArrays, ScriptService scriptService, Client client) {
         super(settings);
         this.bigArrays = bigArrays;
         this.scriptService = scriptService;
+        this.client = client;
+    }
+
+    public Client getClient() {
```

Comment:
hmm any chance we can add a method to `SearchTransportService` that fires off a multisearch via transport service directly? I'd love to not go through a client here?